### PR TITLE
fixes #16868 - properly update pool info

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: app/lib/actions/katello/host/remove_subscriptions.rb [hunk 1]
```diff
@@ -14,16 +14,18 @@ def plan(host, pools_with_quantities_params)
...
-            plan_self(:host_name => host.name)
           end
+          plan_self(:host => host, :host_name => host.name, :pool_ids => pool_ids)
         end
 
         def finalize
-          ::Katello::Pool.where(:id => input[:pool_ids]).each(&:import_data)
-          host = ::Host.find_by(:id => input[:managed][:id])
+          ::Katello::Pool.where(:cp_id => input[:pool_ids]).each(&:import_data)
+          host = ::Host.find_by(:id => input[:host][:id])
```
Comment (jlsherrill) [inline app/lib/actions/katello/host/remove_subscriptions.rb]: why doesn't this work with ListenOnCandlepinevents?

File: app/lib/actions/katello/host/attach_subscriptions.rb [hunk 2]
```diff
@@ -22,13 +22,13 @@ def plan(host, pools_with_quantities_params)
               end
             end
 
-            plan_self(:pool_ids => pool_ids, :host_name => host.name)
+            plan_self(:host => host, :pool_ids => pool_ids, :host_name => host.name)
```
Comment (thomasmckay) [inline app/lib/actions/katello/host/attach_subscriptions.rb]: done here and in remove-subscriptions

File: app/lib/actions/katello/host/remove_subscriptions.rb (hunk 1 above)
Comment (thomasmckay) [inline app/lib/actions/katello/host/remove_subscriptions.rb]: It's a timing thing: The results returned to the UI are not up to date before the candlepin event arrives. This PR updates the local katello data immediately.

Comment (jlsherrill) [inline app/lib/actions/katello/host/remove_subscriptions.rb]: we've had some reports of deadlock issues (such as https://bugzilla.redhat.com/show_bug.cgi?id=1384086) due to this conflicting with the ListenOnCandlepinEvents.  I'm not sure I have a great answer for this, but its something to think about

**Comment by thomasmckay** [comment]
[test]

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**